sqlx/lexer.ts

```diff
@@ -54,15 +55,29 @@ const INNER_SQL_BLOCK_LEXER_TOKEN_NAMES = {
   DOUBLE_QUOTE_STRING: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_doubleQuoteString",
   START_JS_PLACEHOLDER: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_startJsPlaceholder",
   CLOSE_BLOCK: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_closeBlock",
-  ESCAPED_BACKTICK: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_escapedBacktick",
+  BACKSLASH: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_backslash",
   BACKTICK: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_backtick",
-  CAPTURE_EVERYTHING_ELSE: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_captureEverythingElse"
+  CAPTURE_EVERYTHING_ELSE: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_captureEverythingElse",
+  START_QUOTE_SINGLE: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_startQuoteSingle",
+  START_QUOTE_DOUBLE: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_startQuoteDouble"
+};
+
+const INNER_SINGLE_QUOTE_LEXER_TOKEN_NAMES = {
+  BACKSLASH: LEXER_STATE_NAMES.INNER_SINGLE_QUOTE + "_backslash",
+  CLOSE_QUOTE: LEXER_STATE_NAMES.INNER_SINGLE_QUOTE + "_closeQuoteSingle",
+  CAPTURE_EVERYTHING_ELSE: LEXER_STATE_NAMES.INNER_SINGLE_QUOTE + "_captureEverythingElse"
+};
+
+const INNER_DOUBLE_QUOTE_LEXER_TOKEN_NAMES = {
+  BACKSLASH: LEXER_STATE_NAMES.INNER_DOUBLE_QUOTE + "_backslash",
+  CLOSE_QUOTE: LEXER_STATE_NAMES.INNER_DOUBLE_QUOTE + "_closeQuoteDouble",
+  CAPTURE_EVERYTHING_ELSE: LEXER_STATE_NAMES.INNER_DOUBLE_QUOTE + "_captureEverythingElse"
 };
 
 const lexer = moo.states(buildSqlxLexer());
 
 export interface ISyntaxTreeNode {
-  contentType: "sql" | "js" | "jsPlaceholder" | "sqlStatementSeparator" | "sqlComment";
+  contentType: "sql" | "js" | "jsPlaceholder" | "sqlStatementSeparator" | "sqlComment" | "quote";
```

Comment:
i don't think we should make these changes... this implies callers of `constructSyntaxTree` must now handle this case as a separate content type, and (at least for now), quotes should just be contained as part of the `sql` content type.

Reply:
Ah, that does make good sense